lisa/util/__init__.py

```diff
@@ -598,6 +601,58 @@ def is_valid_url(url: str, raise_error: bool = True) -> bool:
     return is_url
 
 
+def is_valid_source_code_package(
```

Comment:
Why is it called "source code package"? What's examples? It looks check a valid url.

Reply:
The function checks for a valid url, expected protocol, expected package origin and expected file name + extension.
 
I did this for the following reasons:
- it's good to enforce https.
- It will help me or other users when we inevitably pass a url in the wrong spot. Getting an error message about the package name or source is better than some confusing message about `failed to make [wrong package name]` where you have to pull the full log and look at the stack trace to see what code called `make`.
- Prevent users from just passing any tarball or git repo to the rdma-core install code and have it 'make install' with sudo.
- Prevent passing an unexpected package type and getting a weird error message about failure to extract it.
 
After getting some feedback from new users I'm trying to intentionally raise more explicit error messages and check assumptions more. `Could not make /path/whatever ` is less useful than `Failed: expected a package like 'https://...rdma-core.vxx.tar.gz and received https://...dpdk.tar.gz`

Reply:
I mean the method is not only used for "source code packages". It can be used for other name patterns. But the method name is called `is_valid_source_code_package`, it's limited, and no examples, no comments, so others are hard to reuse the method.

Reply:
Ah ok, so maybe like `check_url_filename`?